kore/test/Test/Kore/Step/Simplification/AndTerms.hs

```diff
@@ -1242,6 +1210,153 @@ test_functionAnd =
         assertBool "" (Pattern.isSimplified sideRepresentation actual)
     ]
 
+test_Defined :: [TestTree]
+test_Defined =
+    [ testGroup "exact matching" $
+        let partial = Mock.f Mock.a
+            defined = mkDefined partial
+        in
+            [ testCase "\\and(partial, defined)" $ do
+                let expect = [Pattern.fromTermLike partial]
+                (actualAnd, actualUnify) <- simplifyUnify partial defined
+                assertEqual "" expect actualAnd
+                assertEqual "" expect actualUnify
+            , testCase "\\and(defined, partial)" $ do
+                let expect = [Pattern.fromTermLike defined]
+                (actualAnd, actualUnify) <- simplifyUnify defined partial
+                assertEqual "" expect actualAnd
+                assertEqual "" expect actualUnify
```

Comment:
Could you add a comment quickly explaining why we return the pattern on the left here? I assume this has to do with the decision in `equalAndEquals`, right? Or have I gotten this completely wrong?